packages/core/src/routes/verification/index.ts

```diff
@@ -87,8 +87,8 @@ export default function verificationRoutes<T extends UserRouter>(
 
       const user = await queries.users.findUserById(userId);
       const isNewIdentifier =
-        (identifier.type === SignInIdentifier.Email && identifier.value === user.primaryEmail) ||
-        (identifier.type === SignInIdentifier.Phone && identifier.value === user.primaryPhone);
+        (identifier.type === SignInIdentifier.Email && identifier.value !== user.primaryEmail) ||
```

Comment:
no test covered?

Reply:
updated offline, integration tests added in the up coming PR.